### PR TITLE
ENC: better pandas typesetting in ipython nbconvert --to latex (updated)

### PR DESCRIPTION
**Pull request by lmarti**
New pull request with tests and comments. See https://github.com/pydata/pandas/pull/9821 for original PR.

### CODE REVIEW

File: doc/source/whatsnew/v0.16.1.txt [hunk 1]
```diff
@@ -21,6 +21,7 @@ Enhancements
 
 - ``DataFrame.mask()`` and ``Series.mask()`` now support same keywords as ``where`` (:issue:`8801`)
 
+- Added support for rendering DataFrames correctly when converting Jupyter/IPython notebooks to LaTeX (:issue:`9821`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.16.1.txt]: pls move to 0.17.1 whatsnew

File: pandas/core/frame.py [hunk 2]
```diff
@@ -1428,7 +1428,15 @@ def to_latex(self, buf=None, columns=None, col_space=None, colSpace=None,
 
         if buf is None:
             return formatter.buf.getvalue()
-
+    
+    def _repr_latex_(self):
+        """
+        Used for rendering a DataFrame when converting a Jupyter/IPython notebook to LaTeX.
+        For other uses refer to the `to_latex` function.
+        See https://github.com/pydata/pandas/pull/9821 for aditional information.
```
Comment (jreback) [inline pandas/core/frame.py]: just in-line whta is necessary here

File: pandas/tests/test_frame.py [hunk 3]
```diff
@@ -4752,6 +4752,14 @@ def test_repr(self):
...
         self.assertFalse("\r" in repr(df))
         self.assertFalse("a\n" in repr(df))
+    
+    def test_repr_latex(self):
+        # #9821
+        
+        df = DataFrame({'A': [1., 2., 3.],
+                       'B': ['a', 'b', 'c']},
+                       index=np.arange(3))
+        self.assertEqual(df._repl_latex_(), df.to_latex())
```
Comment (jreback) [inline pandas/tests/test_frame.py]: assert the actual string repr. This isn't testing much.

**Comment by jreback** [comment]
closing, pls reopen when updated

File: pandas/core/frame.py (hunk 2 above)
Comment (lmarti) [inline pandas/core/frame.py]: your comment contradicts a previous request.

File: pandas/tests/test_frame.py (hunk 3 above)
Comment (lmarti) [inline pandas/tests/test_frame.py]: This is out of the scope. I havent programmed the repr. Your statement contradicts one of your previous requests.

Comment (jreback) [inline pandas/tests/test_frame.py]: you aren't testing anything then.

File: pandas/core/frame.py (hunk 2 above)
Comment (jreback) [inline pandas/core/frame.py]: how so?

Comment (lmarti) [inline pandas/core/frame.py]: see https://github.com/pydata/pandas/pull/9821

File: pandas/tests/test_frame.py (hunk 3 above)
Comment (lmarti) [inline pandas/tests/test_frame.py]: I know, it was requested at https://github.com/pydata/pandas/pull/9821

**Comment by jreback** [comment]
@lmarti you are being way too terse here and making me work too much.

when you are finally ready for review, pls ping.

**Pull request closed without merging**